extensions/notebook/src/book/bookVersionHandler.ts

```diff
@@ -0,0 +1,86 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { JupyterBookSection, IJupyterBookSectionV1, IJupyterBookSectionV2 } from '../contracts/content';
+import { BookVersion } from './bookModel';
+
+export class BookVersionHandler {
+
+	constructor() { }
+
+	public convertFrom(version: string, tableOfContents: JupyterBookSection): JupyterBookSection {
```

Comment:
convertFrom converts the toc from the provided version to the other? Adding some comments would be great, same for convertTo

Reply:
Oh sorry it's confusing. I'll add comments here.
convertFrom parses a section to JupyterSection, which is the union of both versions. There are conflicting properties between v1 and v2, so I'm converting v1 to v2 while keeping the properties that do not exist in v2.
convertTo is converting a JupyterSection to either v1 or v2.